governance/xc_admin/packages/xc_admin_common/src/price_store.ts

```diff
@@ -0,0 +1,236 @@
+import {
+  MAX_SEED_LENGTH,
+  PublicKey,
+  SystemProgram,
+  TransactionInstruction,
+} from "@solana/web3.js";
+import {
+  MultisigInstruction,
+  MultisigInstructionProgram,
+  UNRECOGNIZED_INSTRUCTION,
+} from "./multisig_transaction";
+import { AnchorAccounts } from "./multisig_transaction/anchor";
+import { PRICE_FEED_OPS_KEY } from "./multisig";
+
+export const PRICE_STORE_PROGRAM_ID: PublicKey = new PublicKey(
+  "3m6sv6HGqEbuyLV84mD7rJn4MAC9LhUa1y1AUNVqcPfr"
+);
+
+export type PriceStoreInitializeInstruction = {
+  payerKey: PublicKey;
+  authorityKey: PublicKey;
+};
+
+export type PriceStoreInitializePublisherInstruction = {
+  authorityKey: PublicKey;
+  publisherKey: PublicKey;
+  bufferKey: PublicKey;
+};
+
+// No need to support SubmitPrices instruction.
+export type PriceStoreInstruction =
+  | {
+      type: "Initialize";
+      data: PriceStoreInitializeInstruction;
+    }
+  | {
+      type: "InitializePublisher";
+      data: PriceStoreInitializePublisherInstruction;
+    };
+
+enum InstructionId {
+  Initialize = 0,
+  SubmitPrices = 1,
+  InitializePublisher = 2,
+}
+
+export function createPriceStoreInstruction(
+  data: PriceStoreInstruction
+): TransactionInstruction {
+  if (data.type == "Initialize") {
```

Comment:
i recommend using switch here, but if not triple equals (`===`)

Reply:
Refactored to use switch.